### PR TITLE
Add network requests allowed API

### PR DESCRIPTION
**Pull request by russelarms**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests?~
* [ ] ~Did you check UI tests on the sample app? They are not executed on CI.~
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Has at least one reviewer tested the PR?

## Description

Add API to block network requests.

## Related PRs or issues

[AB#85221](https://msmobilecenter.visualstudio.com/Mobile-Center/_boards/board/t/Ivan-Team/Backlog%20Items/?workitem=85221)

### CODE REVIEW

File: TestApp/android/app/src/main/java/com/testapp/TestAppNativeModule.java [hunk 1]
```diff
@@ -94,4 +106,12 @@ public void run() {
             }
         });
     }
+
+    @SuppressWarnings("unused")
+    @ReactMethod
+    public void saveIsNetworkAllowed(boolean isAllowed) {
+        mSharedPreferences.edit()
```
[the remark this answers is not in the document]
Comment (DmitriyKirakosyan) [reply TestApp/android/app/src/main/java/com/testapp/TestAppNativeModule.java:113]: Just checked. The state is not saving on the apple SDK side, but it is implemented in Android SDK.
Is it expected?

File: TestApp/ios/Podfile.lock [hunk 2]
```diff
@@ -311,9 +311,10 @@ DEPENDENCIES:
   - Yoga (from `../node_modules/react-native/ReactCommon/yoga`)
 
 SPEC REPOS:
-  trunk:
-    - AppCenter
+  https://github.com/CocoaPods/Specs.git:
```
Comment (MatkovIvan) [inline TestApp/ios/Podfile.lock:314]: Why not keeping trunk for this?

**Review by MatkovIvan — APPROVED** [review]
LGTM

File: TestApp/ios/Podfile.lock (hunk 2 above)
Comment (DmitriyKirakosyan) [reply TestApp/ios/Podfile.lock:314]: Let's keep it as it is for now, as only @russelarms knows the context. We will revert it during the next release.

**Pull request merged**